spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SparkSortStrategy.java

```diff
@@ -119,26 +119,22 @@ public Set<DataFile> rewriteFiles(List<FileScanTask> filesToRewrite) {
       tableCache.add(groupID, table);
       manager.stageTasks(table, groupID, filesToRewrite);
 
-      // Disable Adaptive Query Execution as this may change the output partitioning of our write
-      SparkSession cloneSession = spark.cloneSession();
-      cloneSession.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), false);
-
       // Reset Shuffle Partitions for our sort
       long numOutputFiles =
           numOutputFiles((long) (inputFileSize(filesToRewrite) * sizeEstimateMultiple));
-      cloneSession.conf().set(SQLConf.SHUFFLE_PARTITIONS().key(), Math.max(1, numOutputFiles));
+      spark.conf().set(SQLConf.SHUFFLE_PARTITIONS().key(), Math.max(1, numOutputFiles));
```

Comment:
I'm a little worried that this may be a race condition between various rewrite actions occurring at the same time in the same JVM